### PR TITLE
parse ssh_config for useful data

### PR DESCRIPTION
**Pull request by pxue**
Related: #86 

This PR allows user to pass in a `-config` flag specifying a `ssh_config` file.
I'm using `mikkeloscar/sshconfig`, a fairly simple lexical parser that parses the config syntax and returns host configs.

Purpose of this PR is to allow user to specify different ssh keys, usernames, and other configurables with Sup. For now, only the parsed ssh `User` is used.

One possible concerns might be with parsing large ssh_config files might slow down the initial start time. I've benchmarked a fairly large config file (1k hosts) and saw no significant slow downs.

### CODE REVIEW

**Comment by VojtechVitek** [comment]
This is cool addition, Paul! I'm gonna try it out

**Comment by mingrammer** [comment]
When does support it?

**Comment by mingrammer** [comment]
I can't use ssh connection because current version does not supports custom ssh key ..

**Comment by pkieltyka** [comment]
this would be a nice PR to get working well and merge sometime. As well to solve https://github.com/pressly/sup/issues/86

File: cmd/sup/main.go [hunk 1]
```diff
@@ -246,6 +250,33 @@ func main() {
 		network.Hosts = hosts
 	}
 
+	// --config flag location for ssh_config file
+	if sshConfig != "" {
+		confHosts, err := sshconfig.ParseSSHConfig(sshConfig)
```
Comment (VojtechVitek) [inline cmd/sup/main.go]: This hangs with my `~/.ssh/config`. It enters endless loop and eats 100% CPU.

Comment (VojtechVitek) [reply cmd/sup/main.go]: I take it back... it started working for some reason when I recreated my ~/.ssh/config. I don't get it.

**Pull request merged**